aea/agent.py

```diff
@@ -96,23 +97,31 @@ def __init__(
         :param timeout: the time in (fractions of) seconds to time out an agent between act and react
         :param is_debug: if True, run the agent in debug mode (does not connect the multiplexer).
         :param loop_mode: loop_mode to choose agent run loop.
+        :param runtime: runtime to up agent.
 
         :return: None
         """
         self._identity = identity
         self._connections = connections
 
+        self._runtime = runtime or AsyncRuntime(agent=self, loop=loop)
```

Comment:
See above; I think the default should be `ThreadedRuntime`

Reply:
Not sure how we could provide `AsyncRuntime` as an argument to `agent if it requires agent in its constructor...circularity.

Reply:
I also think `BaseRuntime` needs to have a constructor which allows the following here: `runtime(agent=self, loop=loop)`

Reply:
changed to same behaviour  as for `loop_mode`